drape/glyph_manager.hpp

```diff
@@ -34,7 +35,12 @@ struct TextMetrics
 
   void AddGlyphMetrics(int16_t font, uint16_t glyphId, int32_t xOffset, int32_t yOffset, int32_t xAdvance, int32_t height)
   {
-    m_glyphs.push_back({font, glyphId, xOffset, yOffset, xAdvance});
+    m_glyphs.push_back({{font, glyphId}, xOffset, yOffset, xAdvance});
```

Comment:
nit: emplace_back(GlyphFontAndId(font, glyphId), xOffset, yOffset, xAdvance)

Reply:
A constructor should be defined:
```
FAILED: drape/CMakeFiles/drape.dir/Unity/unity_0_cxx.cxx.o 
/opt/homebrew/bin/ccache /opt/homebrew/bin/ccache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DDEBUG -DQT_CORE_LIB -DUCONFIG_USE_LOCAL -DUSE_FILE32API -I/Users/alex/Developer/omim/omim -I/Users/alex/Developer/omim/omim/cmake-build-debug/3party/gflags/include -I/Users/alex/Developer/omim/omim/3party/glm -I/Users/alex/Developer/omim/omim/3party/opening_hours/. -I/Users/alex/Developer/omim/omim/3party/pugixml/pugixml/src -I/Users/alex/Developer/omim/omim/3party/utfcpp/source -I/Users/alex/Developer/omim/omim/3party/expat/expat/lib -I/Users/alex/Developer/omim/omim/cmake-build-debug/3party/expat/expat -I/Users/alex/Developer/omim/omim/cmake-build-debug/3party/jansson/jansson/include -I/Users/alex/Developer/omim/omim/3party/icu/. -I/Users/alex/Developer/omim/omim/3party/icu/icu/icu4c/source/common -I/Users/alex/Developer/omim/omim/3party/icu/icu/icu4c/source/i18n -I/Users/alex/Developer/omim/omim/3party/protobuf/protobuf/src -I/Users/alex/Developer/omim/omim/3party/freetype -I/Users/alex/Developer/omim/omim/cmake-build-debug/3party/freetype/freetype/include -I/Users/alex/Developer/omim/omim/3party/freetype/freetype/include -I/Users/alex/Developer/omim/omim/3party/vulkan_wrapper/../Vulkan-Headers/include -I/Users/alex/Developer/omim/omim/3party/vulkan_wrapper/. -I/Users/alex/Developer/omim/omim/3party/harfbuzz/harfbuzz/src -isystem /Users/alex/Developer/omim/omim/3party/boost -isystem /opt/homebrew/lib/QtCore.framework/Headers -iframework /opt/homebrew/lib -isystem /opt/homebrew/share/qt/mkspecs/macx-clang -isystem /opt/homebrew/include -g -std=c++20 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -fcolor-diagnostics -fcolor-diagnostics -ffast-math -fno-omit-frame-pointer -Wall -Wextra -Wpedantic -Wno-unused-parameter -MD -MT drape/CMakeFiles/drape.dir/Unity/unity_0_cxx.cxx.o -MF drape/CMakeFiles/drape.dir/Unity/unity_0_cxx.cxx.o.d -o drape/CMakeFiles/drape.dir/Unity/unity_0_cxx.cxx.o -c /Users/alex/Developer/omim/omim/cmake-build-debug/drape/CMakeFiles/drape.dir/Unity/unity_0_cxx.cxx
In file included from /Users/alex/Developer/omim/omim/cmake-build-debug/drape/CMakeFiles/drape.dir/Unity/unity_0_cxx.cxx:4:
In file included from /Users/alex/Developer/omim/omim/drape/attribute_buffer_mutator.cpp:1:
In file included from /Users/alex/Developer/omim/omim/drape/attribute_buffer_mutator.hpp:3:
In file included from /Users/alex/Developer/omim/omim/drape/pointers.hpp:5:
In file included from /Users/alex/Developer/omim/omim/base/assert.hpp:4:
In file included from /Users/alex/Developer/omim/omim/base/internal/message.hpp:3:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/algorithm:1778:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/__algorithm/inplace_merge.h:28:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/__memory/unique_ptr.h:19:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/__memory/allocator_traits.h:304:9: error: no matching function for call to 'construct_at'
        _VSTD::construct_at(__p, _VSTD::forward<_Args>(__args)...);
        ^~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/__config:897:17: note: expanded from macro '_VSTD'
#  define _VSTD std
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/vector:919:21: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<dp::text::GlyphMetrics>>::construct<dp::text::GlyphMetrics, dp::GlyphFontAndId, int &, int &, int &, void, void>' requested here
    __alloc_traits::construct(this->__alloc(), std::__to_address(__tx.__pos_),
                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/vector:1678:9: note: in instantiation of function template specialization 'std::vector<dp::text::GlyphMetrics>::__construct_one_at_end<dp::GlyphFontAndId, int &, int &, int &>' requested here
        __construct_one_at_end(std::forward<_Args>(__args)...);
        ^
/Users/alex/Developer/omim/omim/drape/glyph_manager.hpp:44:14: note: in instantiation of function template specialization 'std::vector<dp::text::GlyphMetrics>::emplace_back<dp::GlyphFontAndId, int &, int &, int &>' requested here
    m_glyphs.emplace_back(GlyphFontAndId{font, glyphId}, xOffset, yOffset, xAdvance);
             ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/include/c++/v1/__memory/construct_at.h:39:38: note: candidate template ignored: substitution failure [with _Tp = dp::text::GlyphMetrics, _Args = <dp::GlyphFontAndId, int &, int &, int &>]: no matching constructor for initialization of 'dp::text::GlyphMetrics'
_LIBCPP_HIDE_FROM_ABI constexpr _Tp* construct_at(_Tp* __location, _Args&&... __args) {
                                     ^
1 error generated.

```